src/cores/ERC721/ERC721Rails.sol

```diff
@@ -21,13 +20,18 @@ import {Initializable} from "../../lib/initializable/Initializable.sol";
 /// @notice This contract implements the Rails pattern to provide enhanced functionality for ERC721 tokens.
 /// @dev ERC721A chosen for only practical solution for large token supply allocations
 contract ERC721Rails is Rails, Ownable, Initializable, TokenMetadata, ERC721, IERC721Rails {
+
+    /// @dev Large batch mints of ERC721A tokens can result in high gas costs upon first transfer of high tokenIds
+    /// To improve UX for token owners unaware of this fact, a mint batch size of 500 is enforced
+    uint256 public constant MAX_MINT_BATCH_SIZE = 500;
```

Comment:
Thoughts on moving this inside `ERC721._mint(to, quantity)` instead? I think it keeps contexts coupled together and is less of a rails issue, and more of a 721A issue.

Reply:
I agree. Thanks for pointing this out. Resolved in be510ff which will be part of the internal/external collapses during the upcoming remove-internal-external PR